flood_api/dependencies/queryparams.py

```diff
@@ -1,23 +1,56 @@
 from datetime import date
-from typing import Annotated
+from typing import Annotated, Optional, Union
```

Comment:
Python 3.10 introduced the `|`-operator for union types: https://peps.python.org/pep-0604/
Instead of `Union[int, float]` you can now write `int | float`. `Optional[MyType]` can also we replaced with `MyType | None`